src/java/org/apache/cassandra/db/partitions/PartitionUpdate.java

```diff
@@ -624,11 +627,11 @@ public void validateIndexedColumns()
 
     public static class PartitionUpdateSerializer
     {
-        private final SchemaManager schemaManager;
+        private final ThrowingFunction<? super TableId, ? extends TableMetadata, ? extends UnknownTableException> tableMetadataResolver;
```

Comment:
out of curiosity - why bounds e.g. for TableId?

Reply:
Obviously it does not matter here. 

It is just like when you know you are always passing an ArrayList to a method, you still make the argument type the most generic as the method can accept, for example a List, or a Collection, or an Iterable. This is exactly the same situation - from PUS point of view, the function can accept just an Object.